### DISCUSSION TITLE
Question about persisting/gathering traefik access logs

### DISCUSSION THREAD

**Discussion by cawoodm** [k3s-io/k3s#5396] [General]
As posted [here](https://community.traefik.io/t/persist-k3s-traefik-logs-to-hostpath/13989) we are using K3S with traefik ingress and would like the access logs persisted to a hostPath volume /var/log/somewhere/traefik/.

In the HelmChartConfig resource we have:

    logs:
      access:
        enabled: true

    volumes:
    - hostPath:
        path: /var/log/somewhere/traefik/
      name: traefik-logs
      type: DirectoryOrCreate
      mountPath: /var/log/traefik/

Expected: The logs are written to the host path /var/log/somewhere/traefik/
Result: No logs are written to the host. Logs are only visible via kubectl logs pod

Apparently traefik is logging to stdout only, can we change this?

10.9.100

Also, it seems the helm chart mounts all volumes as readOnly [link](https://github.com/traefik/traefik-helm-chart/blob/4e53fe085c70211474bbead3cc27727afec1905a/traefik/templates/_podtemplate.tpl#L88) so maybe we are going about this the wrong way.

**Comment by brandond** [comment]
We just bundle the upstream traefik helm chart, so your best bet is to consult the upstream chart and documentation, and continue to work through their forums as you currently are. It sounds like you may need to ask them to update the chart to get a writable volume, as well as add some extra CLI args to set the log filepath.
* https://github.com/traefik/traefik-helm-chart/blob/master/traefik/values.yaml#L178-L188
* https://doc.traefik.io/traefik/observability/logs/#filepath

**Comment by cawoodm** [comment]
According to [this](https://medium.com/trendyol-tech/forwarding-kubernetes-containers-logs-to-elasticsearch-with-fluent-bit-and-showing-logs-with-411587e54e22) blog it should be possible to tail container logs by following `/var/log/containers` or  `/var/log/pods`.

Indeed, we see there logs for every single `kube-system` pod:

## ls /var/log/containers
```
coredns-d76bd69b-2c8r4_kube-system_coredns-***.log
helm-install-traefik-crd-c7dr4_kube-system_helm-***.log
local-path-provisioner-6c79684f77-h67pk_kube-system_local-path-provisioner-***.log
local-path-provisioner-6c79684f77-h67pk_kube-system_local-path-provisioner-***.log
metrics-server-7cd5fcb6b7-f4tpr_kube-system_metrics-server-***.log        
metrics-server-7cd5fcb6b7-f4tpr_kube-system_metrics-server-***.log        
svclb-traefik-z5dbx_kube-system_lb-tcp-443-***.log
svclb-traefik-z5dbx_kube-system_lb-tcp-80-***.log
traefik-7bdbd6f8f5-kt7vm_kube-system_traefik-***.log ✨
```

Just be sure to check all nodes as traefik may not run on all nodes unless you configure it to.
